app/models/goal.py

```diff
@@ -2,4 +2,23 @@
 
 
 class Goal(db.Model):
-    goal_id = db.Column(db.Integer, primary_key=True)
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.Text, nullable=False)
+    tasks = db.relationship("Task", back_populates="goal", lazy=True)
+
+    def to_dict(self):
+        return {
+                "goal": {
+                    "id": self.id,
+                    "title": self.title
+                }
+        }
```

Comment:
Consider stripping off the outer dictionary and returning only the dictionary of key-value pairs. It's not required, but since you only have one key "`goal"` in the outer dictionary and not several to distinguish from, it seems like an extra step to get to the data we want.
```suggestion
        return {
                "id": self.id,
                "title": self.title
            }
```